deeppavlov/evolve.py

```diff
@@ -0,0 +1,322 @@
+        iters = 0
+        result_table = pd.DataFrame(result_table_dict)
+        # write down result table file
+        result_table.loc[:, result_table_columns].to_csv(result_file, index=False, sep='\t')
+
+        log.info("Iteration #{} starts".format(iters))
+        # randomly generate the first population
+        population = evolution.first_generation()
+    else:
+        # if starting evolution from already existing population
+        iters = start_from_population
+        log.info("Iteration #{} starts".format(iters))
+
+        population = []
+        for i in range(population_size):
+            population.append(read_json(expand_path(Path(path_to_population).joinpath(
+                "model_" + str(i)).joinpath("config.json"))))
+            population[i] = evolution.insert_value_or_dict_into_config(
+                population[i], evolution.main_model_path + ["save_path"],
+                str(Path(
+                    evolution.get_value_from_config(evolution.basic_config, evolution.main_model_path + ["save_path"])
+                    ).joinpath(
+                    "population_" + str(start_from_population)).joinpath(
+                    "model_" + str(i)).joinpath(
+                    "model")))
+
+            population[i] = evolution.insert_value_or_dict_into_config(
+                population[i], evolution.main_model_path + ["load_path"],
+                str(Path(
+                    evolution.get_value_from_config(population[i], evolution.main_model_path + ["load_path"]))))
+
+            for path_id, path_ in enumerate(evolution.paths_to_fiton_dicts):
+                population[i] = evolution.insert_value_or_dict_into_config(
+                    population[i], path_ + ["save_path"],
+                    str(Path(evolution.get_value_from_config(evolution.basic_config,
+                                                             evolution.main_model_path + ["save_path"])
+                             ).joinpath("population_" + str(iters)).joinpath("model_" + str(i)).joinpath(
+                        "fitted_model_" + str(path_id))))
+
+            for path_id, path_ in enumerate(evolution.paths_to_fiton_dicts):
+                population[i] = evolution.insert_value_or_dict_into_config(
+                    population[i], path_ + ["load_path"],
+                    str(Path(evolution.get_value_from_config(
+                        population[i], path_ + ["load_path"]))))
+
+    run_population(population, evolution, gpus)
+    population_scores = results_to_table(population, evolution, considered_metrics,
+                                         result_file, result_table_columns)[evolve_metric]
+    log.info("Population scores: {}".format(population_scores))
+    log.info("Iteration #{} was done".format(iters))
+    iters += 1
+
+    while True:
+        if iters == iterations:
+            log.info("End of evolution on iteration #{}".format(iters))
+            break
```

Comment:
What if you set iteration limit to 5 and start from 7th?

Reply:
fixed